.github/workflows/terraform-on-branch-push.yml

```diff
@@ -34,23 +34,17 @@ jobs:
           working-dir: .
           output-file: README.md
           output-method: inject
-          
-      - name: Check for unstaged changes
-        id: git-check
-        run: |
-          sudo chown -R runner:docker .git
-          find . -type f \( -name '*.md' -o -name '*.tf' -o -name '*.tfvars' \) -exec git add {} +
-          git diff --staged --exit-code || echo "changes=true" >> $GITHUB_OUTPUT
-          
-      - name: Commit and Push changes
-        if: steps.git-check.outputs.changes == 'true'
-        run: |
-          git config --global user.name "github-actions[bot]"
-          git config --global user.email "github-actions[bot]@users.noreply.github.com"
-          git commit -m "chore: format terraform code and update docs"
-          git push
-        env:
-          GITHUB_TOKEN: ${{ steps.auth.outputs.token }}
+
+      - name: Commit file updates
+        id: add-commit
+        uses: EndBug/add-and-commit@a94899bca583c204427a224a7af87c02f9b325d5 # v9.1.4
+        with:
+          add: '["**/*.md", "**/*.tf", "**/*.tfvars"]'
+          author_name: github-actions[bot]
+          author_email: github-actions[bot]@users.noreply.github.com
```

Comment:
Do we need to set name and email if we already set default_author that set those property?

<img width="823" height="632" alt="Image" src="https://github.com/user-attachments/assets/e79785b6-36e4-4eae-88a8-8932ce749ded" />